src/datasets/utils/file_utils.py

```diff
@@ -229,17 +229,28 @@ class DownloadConfig:
     """
 
     cache_dir: Optional[Union[str, Path]] = None
-    force_download: bool = False
+    force_download: Optional[bool] = None  # default False
+    _is_force_download_default: bool = False
```

Comment:
Why do you need this ?

Reply:
This has to do with the fact of current over-use of `DownloadConfig` for many different tasks/concerns...

Reply:
I see !
What if the value `force_download` is updated during `download_and_prepare` ? In this case `_is_force_download_default` would still be True even though it's not the default value "None" anymore.
I find the name of the attribute confusing in this case

Reply:
I have renamed the attributes, e.g. `_is_force_download_set_by_user`.